### PR TITLE
Minor changes in DTOs and validation, API Tags

### PR DESCRIPTION
**Pull request by shivambhatia5432**
**Description**

Minor changes in DTOs and validation, API Tags

**Related Issue** 
- issue goes here

Fixes # [ISSUE]

- [x] Code
- [ ] User Interface
- [ ] New Feature
- [ ] Documentation
- [ ] Testing

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My UI is responsive
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [ ] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Review by Devesh21700Kumar — COMMENTED** [review]
Hey @shivambhatia5432 
Actually, the validation pipe is not being set up globally and is only being used in instances such as the creation of a new object, or creations of a user in other endpoint sets of controllers for other schemas.

So I think you may remove those changes to the main.ts file.

**Comment by shivambhatia5432** [comment]
Done @Devesh21700Kumar thank you so much man for helping

**Comment by Abhishek-kumar09** [comment]
Please resolve merge conflicts.

**Comment by shivambhatia5432** [comment]
> Please resolve merge conflicts.

Done

File: src/modules/assignment/assignment.controller.ts [hunk 1]
```diff
@@ -12,23 +12,27 @@ import {
...
+import { ApiCreatedResponse, ApiProperty, ApiTags } from '@nestjs/swagger';
 import { UpdateAssignmentDTO } from './dto/update-assignment.dto';
+import { IsEmail, Length } from 'class-validator';
 
 class AssignmentResponseBody {
   @ApiProperty({ required: true, example: '605e3fd9acc33583fb389aec' })
   id: string;
 
   @ApiProperty({ required: true, example: 'Noob' })
+  @Length(3, 20)
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:24]: The class validators will have no effect here, Please remove the class validators from the swagger response body.

File: src/modules/assignment/assignment.controller.ts [hunk 2]
```diff
@@ -12,23 +12,27 @@ import {
...
 import { AssignmentService } from './assignment.service'; //eslint-disable-line 
 import { CreateAssignmentDTO } from './dto/create-assignment.dto'; //eslint-disable-line 
-import { ApiCreatedResponse, ApiProperty } from '@nestjs/swagger';
+import { ApiCreatedResponse, ApiProperty, ApiTags } from '@nestjs/swagger';
 import { UpdateAssignmentDTO } from './dto/update-assignment.dto';
+import { IsEmail, Length } from 'class-validator';
 
 class AssignmentResponseBody {
   @ApiProperty({ required: true, example: '605e3fd9acc33583fb389aec' })
   id: string;
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:21]: `id` to `_id`

File: src/modules/assignment/assignment.controller.ts [hunk 3]
```diff
@@ -12,23 +12,27 @@ import {
...
 
   @ApiProperty({ required: true, example: 'Noob' })
+  @Length(3, 20)
   name: string;
 
-  @ApiProperty({ required: true, example: 'Coder' })
+  @ApiProperty({ required: true, example: 'www.google.com' })
   link: string;
 
   @ApiProperty({ required: true, example: 'noobcoder@gmai.com' })
```
Comment (Abhishek-kumar09) [inline src/modules/assignment/assignment.controller.ts:30]: `noobcoder@gmail.com`

File: src/modules/chat/chat.controller.ts [hunk 4]
```diff
@@ -10,24 +10,23 @@ import {
...
+import { ApiCreatedResponse, ApiProperty, ApiTags } from '@nestjs/swagger';
 class ChatResponseBody {
   @ApiProperty({ required: true, example: '605e3fd9acc33583fb389aec' })
   id: string;
 
   @ApiProperty({ required: true, example: 'John' })
   sender: string;
-
-  @ApiProperty({ required: true, example: 'Johny' })
+  @ApiProperty({ required: true, example: 'noob2' })
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:22]: please revert this.

File: src/modules/chat/chat.controller.ts [hunk 5]
```diff
@@ -10,24 +10,23 @@ import {
   UsePipes,
   ValidationPipe,
 } from '@nestjs/common';
-import { ApiCreatedResponse, ApiProperty } from '@nestjs/swagger';
 import { ChatService } from './chat.service'; //eslint-disable-line 
 import { CreateChatDTO } from './dto/create-chat.dto'; //eslint-disable-line 
-
+import { ApiCreatedResponse, ApiProperty, ApiTags } from '@nestjs/swagger';
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:15]: Please keep that line on the same linenumber

File: src/modules/user/user.controller.ts [hunk 6]
```diff
@@ -22,24 +23,29 @@ class UserResponseBody {
...
   @ApiProperty({ required: true, example: 'Noob' })
+  @Length(3, 20)
   first_name: string;
 
   @ApiProperty({ required: true, example: 'Coder' })
+  @Length(3, 20)
   last_name: string;
 
   @ApiProperty({ required: true, example: 'noobcoder@gmai.com' })
+  @IsEmail()
```
Comment (Abhishek-kumar09) [inline src/modules/user/user.controller.ts:34]: These classvalidators have no effect on the doc I guess

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
It gone unnoticed during previous review but please change `submit_by` to `User` that will contain user's `_id`
https://github.com/codeforcauseorg/edu-server/blob/55c096e96d99e2fb2fa319662be9d1e1f019b1bd/src/schemas/assignment.schema.ts#L18

**Comment by shivambhatia5432** [comment]
> It gone unnoticed during previous review but please change `submit_by` to `User` that will contain user's `_id`
> https://github.com/codeforcauseorg/edu-server/blob/55c096e96d99e2fb2fa319662be9d1e1f019b1bd/src/schemas/assignment.schema.ts#L18

What is submit_by format in actual? In assignment.schema it is number and in assignment.controller it is string

**Comment by Abhishek-kumar09** [comment]
> > It gone unnoticed during previous review but please change `submit_by` to `User` that will contain user's `_id`
> > https://github.com/codeforcauseorg/edu-server/blob/55c096e96d99e2fb2fa319662be9d1e1f019b1bd/src/schemas/assignment.schema.ts#L18
> 
> What is submit_by format in actual? In assignment.schema it is number and in assignment.controller it is string


It will be actually mongoId. so it should be `string`. You can decorate that with `@IsMongoId` in the Dto.

**Pull request closed without merging**